config/src/main/java/com/redhat/lightblue/config/LightblueFactory.java

```diff
@@ -173,6 +174,8 @@ private synchronized void initializeMetadata() throws IOException, ClassNotFound
             getJsonTranslator().setValidation(EntityInfo.class, cfg.isValidateRequests());
 
             metadata = cfg.createMetadata(datasources, getJSONParser(), this);
+            initializeFactory(); //re enforce the factory initialization
```

Comment:
You mean replace the two lines (the initializeFactory and the hook init lines) to use the getFactory() directly so it may initialize if it is needed, right?
I thought that when I code these lines but I thought it would make the requirements implicit. I even commented to force the attention that the factory is really needed, so any change in the factory init could impact this method.

But if you think it would be better using that single line statement, I can update the PR immediately after you confirm this.